src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -727,21 +701,17 @@ void c_typecheck_baset::typecheck_expr_operands(exprt &expr)
      expr.get(ID_statement)==ID_function_call)
   {
     // don't do function operand
-    assert(expr.operands().size()==2);
-
-    typecheck_expr(expr.op1()); // arguments
+    typecheck_expr(to_binary_expr(expr).op1()); // arguments
   }
   else if(expr.id()==ID_side_effect &&
           expr.get(ID_statement)==ID_statement_expression)
   {
-    typecheck_code(to_code(expr.op0()));
+    typecheck_code(to_code(to_unary_expr(expr).op()));
```

Comment:
We should have `side_effect_statement_expressiont` it seems?!

Reply:
yes, added